### ISSUE TITLE
Crash on windows 11.

### ISSUE THREAD

**Issue by sj0n** [chapar-rest/chapar#26]
**Describe the bug**
Launching the binary crash the program immediately.

**To Reproduce**
1. I try running go run main.go and it returns `"2024/05/14 17:45:29 open C:\Users\User\AppData\Roaming/chapar/config.yaml: The system cannot find the path specified.
exit status 1"`.
2. I created the folder manually and it returns `"2024/05/14 17:48:56 open C:\Users\User\AppData\Roaming/chapar/C:\Users\User\AppData\Roaming\chapar\Default Workspace/preferences/preferences.yaml: The filename, directory name, or volume label syntax is incorrect.
exit status 1"`.

**Expected behavior**
Launch the application.

**Screenshots**
![image](https://github.com/chapar-rest/chapar/assets/21690823/f9a5acec-523e-4268-b3a4-15f3dc686271)

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 11

**Comment by mirzakhany** [comment]
Thanks for reporting this. It seems that app is building a wrong path on boot up for configuring directory on windows 🤦🏻‍♂️

I’ll fix soon

**Comment by RafaelRCL** [comment]
So from diving a little bit in  the code I found the culprits

on  `filesystem.go ln 79` you are using:

```go
dir, err := GetConfigDir() 
``` 
* This line is the one giving the error @sj0n was having, and I had aswell.
* * 

while on other places you were using 
```go
dir, err := CreateConfigDir()
```

After fixing this I found another windows bug, still on the same file.

on `filesystem.go ln 149` you are using
``` go
if _, err := os.Stat(filePath); os.IsNotExist(err) {
		ws := domain.NewWorkspace(path.Base(dirPath))
		ws.FilePath = filePath
		if err := SaveToYaml(filePath, ws); err != nil {
			return nil, err
		}

		return ws, nil
	}
```

The correct package to use for filepaths is `path/filepath` which is described in its [documentation](https://pkg.go.dev/path/filepath)
>Package filepath implements utility routines for manipulating filename paths in a way compatible with the target operating system-defined file paths.
The filepath package uses either forward slashes or backslashes, depending on the operating system. To process paths such as URLs that always use forward slashes regardless of the operating system, see the [path](https://pkg.go.dev/path) package.The filepath package uses either forward slashes or backslashes, depending on the operating system. To process paths such as URLs that always use forward slashes regardless of the operating system, see the [path](https://pkg.go.dev/path) package.

``` go
if _, err := os.Stat(filePath); os.IsNotExist(err) {
		ws := domain.NewWorkspace(filepath.Base(dirPath)) // which need to be imported as path/filepath
		ws.FilePath = filePath
		if err := SaveToYaml(filePath, ws); err != nil {
			return nil, err
		}

		return ws, nil
	}
```

Package filepath implements utility routines for manipulating filename paths in a way compatible with the target operating system-defined file paths.

After doing those 2 changes I was able to run in windows. I don't know if there are any other bugs while using since I only ran after making these changes.

I can make a pull requets, but since there were no tests for these files I'll let you decide what to do because I don't have Linux and MacOS machines with GUI available to test the first change I proposed
